### PR TITLE
Change: remove in-app titlebar

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
The connection problems are now indicated by a badge on the sidebar connections icon and listed in the sidebar pane.

image

image

**Description**
On Windows and Linux we are not able to overlay native window controls
into the app and omit the system titlebar. This creates a "dual bar"
appearance with the system titlebar and our in-app toolbar. This dual
bar takes up screen space with little value - only the current connection string and problems are placed there.

This change removes the in-app toolbar and leaves the native window
titlebar and controls. It also moved the problems into the connection
sidebar pane and adds badges to the connection sidebar icon when there are problems with the connection.

Fixes: #1540

### CODE REVIEW

File: packages/studio-base/src/components/DocumentTitleAdapter.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+export default function DocumentTitleAdapter(): JSX.Element {
+  const { currentSourceName } = usePlayerSelection();
+
+  useEffect(() => {
+    if (!currentSourceName) {
+      window.document.title = "Foxglove Studio";
+      return;
+    }
+    window.document.title = `Foxglove Studio - ${currentSourceName}`;
```
Comment (amacneil) [inline packages/studio-base/src/components/DocumentTitleAdapter.tsx:20]: can you change to `${currentSourceName} - Foxglove Studio` so it's consistent with other applications.

if you want to be more specific, macOS desktop should only show `currentSourceName` and we should append ` - Foxglove Studio` everywhere else

Comment (defunctzombie) [reply packages/studio-base/src/components/DocumentTitleAdapter.tsx:20]: https://github.com/foxglove/studio/pull/1678

File: desktop/main/StudioWindow.ts [hunk 2]
```diff
@@ -115,7 +111,7 @@ function newStudioWindow(deepLinks: string[] = []): BrowserWindow {
   browserWindow.webContents.on("ipc-message", (_event: unknown, channel: string) => {
     if (channel === "window.toolbar-double-clicked") {
       const action: string =
-        systemPreferences.getUserDefault?.("AppleActionOnDoubleClick", "string") || "Maximize";
+        systemPreferences.getUserDefault?.("AppleActionOnDoubleClick", "string") ?? "Maximize";
```
Comment (amacneil) [inline desktop/main/StudioWindow.ts:114]: are you sure that `getUserDefault` returns undefined rather than empty string?

Comment (defunctzombie) [reply desktop/main/StudioWindow.ts:114]: https://github.com/foxglove/studio/pull/1680

**Pull request merged**